components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/BuildTimeMetricsFields.java

```diff
@@ -0,0 +1,48 @@
+package com.gradle.enterprise.cli;
+
+import com.gradle.enterprise.model.BuildTimeMetrics;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public enum BuildTimeMetricsFields {
+    // The order the enums are defined controls the order the fields are printed in the CSV
+    INITIAL_BUILD_TIME("Initial Build Time", d -> formatDuration(d.initialBuildTime)),
+    INSTANT_SAVINGS("Instant Savings", d -> formatDuration(d.instantSavings)),
+    INSTANT_SAVINGS_BUILD_TIME("Instant Savings Build Time", d -> formatDuration(d.instantSavingsBuildTime)),
+    PENDING_SAVINGS("Pending Savings", d -> formatDuration(d.pendingSavings)),
+    PENDING_SAVINGS_BUILD_TIME("Pending Savings Build Time", d -> formatDuration(d.pendingSavingsBuildTime)),
+    ;
+
+    public final String label;
+    public final Function<BuildTimeMetrics, String> value;
+
+    BuildTimeMetricsFields(String label, Function<BuildTimeMetrics, String> value) {
+        this.label = label;
+        this.value = value;
+    }
+
+    public static Stream<BuildTimeMetricsFields> ordered() {
+        return Arrays.stream(BuildTimeMetricsFields.values());
+    }
+
+    private static String formatDuration(Duration duration) {
```

Comment:
The only downside to doing this here is that we're now putting some of the display logic back into the Java code, where we used to have it only in the bash scripts.  I'm not saying we should undo this, just commenting that the separation of responsibilities is a bit muddy.

Reply:
Yes, the trend is and will be to have more in the Java code and less in the Bash scripts.